task-8/src/api/controllers/cart.controller.ts

```diff
@@ -8,64 +8,86 @@ import { cartService } from "../services";
+  } catch (e) {
+    next(e);
+  }
 });
 
-cartController.put("/", (req, res) => {
-  const reqUserId = req.headers[AUTH_TOKEN_HEADER] as string;
-  const requestBody = req.body;
-  const validatedRequest = updateCartSchema.validate({
-    userId: reqUserId,
-    ...requestBody,
-  });
-
-  // Request is not valid
-  if (validatedRequest.error) {
-    throw new APIError(
-      validatedRequest.error.message,
-      STATUS_CODES.BAD_REQUEST
+cartController.put("/", async (req, res, next) => {
+  try {
+    const reqUserId = req.headers[AUTH_TOKEN_HEADER] as string;
+    const requestBody = req.body;
+    const validatedRequest = updateCartSchema.validate({
+      userId: reqUserId,
+      ...requestBody,
+    });
+
+    // Request is not valid
+    if (validatedRequest.error) {
+      throw new APIError(
+        validatedRequest.error.message,
+        STATUS_CODES.BAD_REQUEST
+      );
+    }
+
+    const { userId, productId, count } = validatedRequest.value;
+    const updateCartResult = await cartService.updateCart(
+      userId,
+      productId,
+      count
     );
-  }
 
-  const { userId, productId, count } = validatedRequest.value;
-  const updateCartResult = cartService.updateCart(userId, productId, count);
+    // No product with such ID
+    if (updateCartResult === cartService.CART_ERRORS.NO_PRODUCT) {
+      throw new APIError("Products are not valid", STATUS_CODES.BAD_REQUEST);
+    }
 
-  // No product with such ID
-  if (updateCartResult === cartService.CART_ERRORS.NO_PRODUCT) {
-    throw new APIError("Products are not valid", STATUS_CODES.BAD_REQUEST);
-  }
+    // No cart exist for user
+    if (updateCartResult === cartService.CART_ERRORS.NO_CART) {
+      throw new APIError("Cart was not found", STATUS_CODES.NOT_FOUND);
+    }
 
-  // No cart exist for user
-  if (updateCartResult === cartService.CART_ERRORS.NO_CART) {
-    throw new APIError("Cart was not found", STATUS_CODES.NOT_FOUND);
+    const { cart, total } = updateCartResult;
+    res.status(STATUS_CODES.OK).json(generateResponse({ cart, total }));
+  } catch (e) {
+    next(e);
   }
-
-  const { cart, total } = updateCartResult;
-  res.status(STATUS_CODES.OK).send(generateResponse({ cart, total }));
 });
 
-cartController.delete("/", (req, res) => {
-  const userId = req.headers[AUTH_TOKEN_HEADER] as string;
+cartController.delete("/", async (req, res, next) => {
+  try {
+    const userId = req.headers[AUTH_TOKEN_HEADER] as string;
 
-  cartService.emptyCartByUserId(userId);
+    await cartService.emptyCartByUserId(userId);
 
-  res.status(STATUS_CODES.OK).send(generateResponse({ success: true }));
+    res.status(STATUS_CODES.OK).json(generateResponse({ success: true }));
+  } catch (e) {
+    next(e);
+  }
 });
 
-cartController.post("/checkout", (req, res) => {
-  const userId = req.headers[AUTH_TOKEN_HEADER] as string;
+cartController.post("/checkout", async (req, res, next) => {
+  try {
+    const userId = req.headers[AUTH_TOKEN_HEADER] as string;
```

Comment:
tip: you can create a middleware to extract userId from request headers.
This approach makes your code easier to maintain and update since you only need to modify the middleware in one place if changes are required. And by extracting data from headers in a middleware function, you keep this logic separate from the actual route handlers, which focus on handling business logic.

Reply:
Done in scope of the next ticket as a part of Authorization middleware which set's JWT token payload into `user` field of request.
Please follow: #11 -> task-9/src/api/middlewares/auth.middleware.ts